Update custom to generic terminology (#2265)
hashicorp/design-system: website/docs/components/side-nav/partials/guidelines/guidelines.md

![User dropdown menu in side-nav's header](/assets/components/side-nav/user-dropdown.png)

### Generic content

> The `generic` type supports any generic content, local components, or Helios components within the header via an instance swap property (genericInstance) in Figma.

Reviewer: [polish] Aligning to how backticks are used elsewhere.

```suggestion
The generic type supports any generic content, local components, or Helios components within the header via an instance swap property (`genericInstance`) in Figma.
```

Use external links sparingly. Avoid using this property to link pages that are unrelated to the product's navigation.
!!!

### Generic content

> Toggle `hasGenericContent` on to support any additional generic content, local components, or Helios components within the body container via an instance swap property (genericInstance) in Figma.

Reviewer: @jorytindall @andgen404 These may have been reverted when you merged up your branch. They still say `hasTopCustomContent`, `topCustomContent`, `hasBottomCustomContent`, and `bottomCustomContent` in Figma.

```suggestion
The topGenericInstance and bottomGenericInstance properties support any additional generic content, local components, or Helios components within the body container via instance swap properties (`topGenericInstance`, `bottomGenericInstance`) in Figma.
```

Author: Ah, yeah. I made note of this. Thank you! @heatherlarsen
